src/hooks/useStockState.ts

```diff
@@ -4,7 +4,8 @@ import { StockContext } from '../components/StockContext';
 import { Stock } from './useStock';
 import { useStockValue } from './useStockValue';
 
-export type SetAction<V> = (value: V) => void;
+export type SetStateAction<S> = S | ((value: S) => S);
+type Dispatch<A> = (value: A) => void;
```

Comment:
These lines should be inside `types` folder, not here

Reply:
Should I move Dispatch type in single file?